src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_lineaire.cpp

```diff
@@ -182,522 +97,52 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeLineaire(PROBLEME_HEBDO* pro
         CORRESPONDANCES_DES_CONTRAINTES& CorrespondanceCntNativesCntOptim
             = problemeHebdo->CorrespondanceCntNativesCntOptim[pdt];
 
+        AreaBalance areaBalance(problemeHebdo);
```

Comment:
We can see that we are hampered by the order of constraints insertion in the problem.
- It makes the current source file more complex than it should because we have to be careful with the loop over time step, areas, ... the constraint is built in.
- We mix insertion order and constraint building

Could we not separate constraint insertion order and building ?
So we could create and use constraints with more flexibility.
Like pointed in [this ticket](https://github.com/AntaresSimulatorTeam/Antares_Simulator/issues/1612),